File: solutions/4138-minimum-deletion-cost-to-make-all-characters-equal/solution.py
class Solution(object):
    def minCost(self, s, cost):
        """
        :type s: str
        :type cost: List[int]
        :rtype: int
        """
        d={}
        for i in range(len(s)):
            if s[i] in d:
                d[s[i]]+=cost[i]
            else:
                d[s[i]]=cost[i]
        d=sorted(d.values())
        return sum(d[:-1])

File: solutions/4138-minimum-deletion-cost-to-make-all-characters-equal/solution.java
class Solution {
    public long minCost(String s, int[] cost) {
        long arr[]=new long[26];
        long sum=0,max=0;
        int i=0;
        for(char a:s.toCharArray())
        {
            arr[a-'a']+=cost[i];
            sum+=cost[i];
            max=Math.max(arr[a-'a'],max);
            i++;
        }
        return (long)(sum-max);
    }
}
